### PR TITLE
Onboarding medium importer

### PR DESCRIPTION
**Pull request by zaerl**
#### Changes proposed in this Pull Request

* Add a Medium importer to onboarding flow, see: TAK5FpwTvfbENJDfHc7Aq1-fi
* New `ImporterDrag` component, which can be used with all the import flow that necessitates a file upload.
* Minor refactoring of `client/lib/importer/importer-config.js`

#### Testing instructions

* Go to https://medium.com/me/export
* Click "Export"
* Medium will send you a .zip with the exported site (it could take several minutes, depending on the size of the site)
* Enable the **gutenboarding/import-from-medium** feature flag (already enabled on development environment)
* Go to http://calypso.localhost:3000/start/importer/capture?siteSlug=YOUR-SITE
* Insert the Medium site URL
* Select the `.zip` file you have just downloaded

**Important notice**: Medium sends an email with a signed S3 URL. The file is wrongly uploaded to S3 with `text/html` MIME type instead of `application/zip`, so the browser could not start the download automatically. In this case, you need to force the download as `.zip`. Maybe it's a temporary bug.

This component will incorporate the changes on #59238.

### CODE REVIEW

File: client/signup/steps/import-from/medium/index.tsx
Comment (a8ci18n) [inline client/signup/steps/import-from/medium/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Importing content from Medium' )` *ES Score: 9*

**Comment by zaerl** [comment]
@bogiii @vishnugopal, once we create a good base for a file-upload importer flow (all the importers except **Wix**), everything will be incorporated quickly because of the differences are only some strings and nothing more.

See Medium custom ones at https://github.com/Automattic/wp-calypso/blob/33df2d1ace7956e701e5980ab2bd47cb4bb859b1/client/signup/steps/import-from/medium/index.tsx#L86-L114).

**Comment by bogiii** [comment]
<table>
<tr>
<td  colspan="2">Screenshot 2021-12-22 at 13 11 27</td>
</tr>
<tr>
<td>
Screenshot 2021-12-22 at 13 45 35
</td>
<td>
Screenshot 2021-12-22 at 13 45 13
</td>
</tr>
<tr>
<td>
Screenshot 2021-12-22 at 13 49 42
</td>
<td>
Screenshot 2021-12-22 at 13 49 52
</td>
</tr>
</table>

I'll leave the screenshot here to discuss and decide if we can go with the current state and fix it in the next iterations.

@vishnugopal what do you think?

File: client/signup/steps/import-from/medium/index.tsx
Comment (bogiii) [inline client/signup/steps/import-from/medium/index.tsx:43]: In the meantime, I've improved the way of deciding when to run the import. Maybe you can reuse it here.
https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/import-from/wix/index.tsx#L37-L74

File: client/signup/steps/import-from/medium/index.tsx
Comment (bogiii) [inline client/signup/steps/import-from/medium/index.tsx:49]: It's related to the comment about improvement on the running import job.

Comment (zaerl) [reply client/signup/steps/import-from/medium/index.tsx:49]: That's ok. Maybe we can have a reusable class for all the file-upload import flows. But I think that is an optimization that we can do in the next iteration.

**Review by vishnugopal — APPROVED** [review]
Assorted: Observations

* Am OK with styles waiting until next PR, and designer's feedback before merge.
* Please toggle `"gutenboarding/import-from-medium": true` at least in development.json to make testing easier.
* We can provide any medium page, and any other export as far as I can see. I tested with `medium.com` for example. This is OK for now.
* The indeterminate spinner definitely looks odd: 
CleanShot 2021-12-23 at 12 58 11@2x can we change to neutral (white/brown) colours?
* Otherwise things seem to work! Good stuff!

**Comment by vishnugopal** [comment]
So let us:

- [ ] Change the things that can be done with CSS
- [ ] Make a list of things that cannot be done with CSS

File: client/signup/steps/import-from/medium/index.tsx
Comment (bogiii) [reply client/signup/steps/import-from/medium/index.tsx:49]: Agree with you!

**Comment by zaerl** [comment]
Made the first changes:

image

File: client/signup/steps/import-from/components/importer-drag/style.scss
Comment (bogiii) [inline client/signup/steps/import-from/components/importer-drag/style.scss]: This could be: `var( --studio-gray-100 )`

File: client/signup/steps/import-from/components/importer-drag/style.scss
Comment (bogiii) [inline client/signup/steps/import-from/components/importer-drag/style.scss]: Also, `#c3c4c7` -> `var( --studio-gray-10 )`

File: client/signup/steps/import-from/components/importer-drag/style.scss
Comment (bogiii) [inline client/signup/steps/import-from/components/importer-drag/style.scss]: `#101517` -> `--studio-gray-100`

**Review by bogiii — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7056337

Hi @zaerl, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Import content from %(importerName)s`
- `Import your posts, tags, images, and videos from your %(importerName)s export file`

Thank you in advance!

**Comment by zaerl** [comment]
@a8ci18n check this new comment: https://github.com/Automattic/wp-calypso/pull/59436#issuecomment-1000315844. You can find these strings there.

These new strings will be used for all the importers. So `%(importerName)s` is one of:

1. WordPress
2. Blogger
3. **Medium** (this one)
4. Substack
5. Squarespace

Thanks! cc @vishnugopal @bogiii

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7056337).

**Pull request merged**
